content/en/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md

```diff
@@ -1,76 +1,201 @@
+Figure 1 illustrates the architecture of the resource metrics pipeline.
+
+{{< mermaid >}}
+flowchart RL
+subgraph cluster[Cluster]
+direction RL
+S[ <br><br> ]
+A[Metrics-<br>Server]
+subgraph B[Nodes]
+direction TB
+D[cAdvisor] --> C[kubelet]
+E[Container<br>runtime] --> D
+E1[Container<br>runtime] --> D
+P[pod data] -.- C
+end
+L[API<br>server]
+W[HPA]
+C ---->|Summary<br>API| A -->|metrics<br>API| L --> W
+end
+L ---> K[kubectl<br>top]
+classDef box fill:#fff,stroke:#000,stroke-width:1px,color:#000;
+class W,B,P,K,cluster,D,E,E1 box
+classDef spacewhite fill:#ffffff,stroke:#fff,stroke-width:0px,color:#000
+class S spacewhite
+classDef k8s fill:#326ce5,stroke:#fff,stroke-width:1px,color:#fff;
+class A,L,C k8s
+{{< /mermaid >}}
+
+Figure 1. Resource Metrics Pipeline
+
+The architecture components, from right to left in the figure, consist of the following:
+
+* [cAdvisor](https://github.com/google/cadvisor) - Daemon for collecting, aggregating and exposing Docker container metrics. Note that non-Docker container runtimes that support metrics collection through a [CRI will not require cAdvisor](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-node/cri-container-stats.md).
+* [kubelet](https://kubernetes.io/docs/concepts/overview/components/#kubelet) - Node agent for managing container resources. Resource metrics are accessible using the `/metrics/resource` and `/stats` kubelet API endpoints. 
+* [Summary API](#summary-api-source) - Kubelet API for discovering and retrieving per-node summarized stats available through the `/stats` endpoint. 
+* [metrics-server](#metrics-server) - Cluster addon component that collects and aggregates resource metrics pulled from each kubelet. The K8s API server serves up the Metric API endpoints for use by HPA, VPA and the `kubectl top` command.
+* [Metrics API](#metrics-api) - Kubernetes API supporting access to CPU and memory used for workload autoscaling. Metrics Server is the default implementation provided with many popular K8s distributions, however it can be replaced by alternative adapters based on your preferred monitoring solution.
+  
+<!-- body -->
 
-The API is defined in [k8s.io/metrics](https://github.com/kubernetes/metrics/blob/master/pkg/apis/metrics/v1beta1/types.go)
-repository. You can find more information about the API there.
+## Metrics API
+
+The metrics-server implements the Metrics API. This API allows you to access CPU and memory usage for the nodes and pods in your cluster. Its primary role is to feed resource usage metrics to K8s autoscaler components. 
+
+Here is an example of the Metrics API request for a `minikube` node piped through `jq` for easier reading:
+```text
+kubectl get --raw "/apis/metrics.k8s.io/v1beta1/nodes/minikube" | jq '.'
+```
+Here is the same API call using `curl`:
+```text
+curl http://localhost:8080/apis/metrics.k8s.io/v1beta1/nodes/minikube
+```
+Sample reply:
+```text
```

Comment:
nit: this is JSON, not text.

Reply:
will do in separate commit to test locally